src/OSPSuite.Core/Domain/Model.cs

```diff
@@ -60,16 +63,23 @@ public IContainer Neighborhoods
 
       public string MoleculeNameFor(string quantityPath)
       {
-         var quantity = Root?.EntityAt<IQuantity>(quantityPath.ToPathArray());
-         return MoleculeNameFor(quantity);
+         var entity = Root?.EntityAt<IEntity>(quantityPath.ToPathArray());
+         return MoleculeNameFor(entity);
       }
 
-      public string MoleculeNameFor(IQuantity quantity)
+      public string MoleculeNameFor(IEntity entity)
       {
-         if (quantity == null)
+         if (entity == null)
             return string.Empty;
 
-         return quantity.IsAnImplementationOf<IMoleculeAmount>() ? quantity.Name : quantity.ParentContainer?.Name;
+         switch (entity)
+         {
+            case IMoleculeAmount _:
+            case IContainer container when container.ContainerType == ContainerType.Molecule:
+               return entity.Name;
+         }
+
+         return MoleculeNameFor(entity.ParentContainer);
```

Comment:
@Yuri05 Instead of your proposed solution, it feels like we can make it a big more generic by traversing the tree recursively and returning the name of the first container (including self) that is either a MoleculeAmount or a Container of type Molecule. 
Thoughts?

Reply:
Probably OK